### ISSUE TITLE
[Bug]: Language and region url parameters and request headers aren’t following user settings

### ISSUE THREAD

**Issue by hollownights** [FreeTubeApp/FreeTube#3745]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

I have the “United Kingdom” set as “Region for Trending” and “English (UK)” set as “Language preference”, but when I type something in the search bar, FreeTube sends a request for suggestqueries.google.com with parameters “&hl=en&gl=US”, getting the language right (hl=en), but not the region (gl=US). 

That request seems to be made with `“accept-language: *”` as a request header, but then, when trying to access anything, be it opening a channel’s page or loading a video, things get wonky when it comes to the “accept-language” parameter: “Image” requests seem to all be made with my system’s language (which isn’t english) as the parameter’s value, but “Fetch/XHR” requests get divided between `“*”`, “en-US” and my system’s language as value, with requests to “iframe_api” and “videoplayback” (to get a video) always being made with my system’s language as the parameter’s value.

This behavior is really problematic because:
1) It reduces privacy by providing more data points for fingerprinting: When accessing YouTube through a web browser, YouTube would see only one language as value for “accept-language” and the URL parameters would also most likely match it. FreeTube is providing not one, but tree (counting “*”) languages as values;
2) It messes up with content recommendation: FreeTube is not always using the region the user wants to see (the UK, for me) and all the other languages informed (like the system’s language) probably aren’t helping either;
3) It (probably) messes up with subtitles selection; 
4) It (probably) plays a part in issues like #3742 and #2419, selecting the wrong audio language (I’m facing this Tower of Babel nightmare while using nightly 3089).

### Expected Behavior

Request headers and URL parameters for region and language should always follow user settings.

### Issue Labels

accessibility issue, API issue, inconsistent behavior, usability issue

### FreeTube Version

nightly 3089

### Operating System Version

Windows 10

### Installation Method

.exe

### Primary API used

Local API

### Nightly Build

- [X] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by absidue** [comment]
Unfortunately we can't do that yet, as for things like video lists (channels, subscriptions, trending, search results, watch next feed) only gives us relative published dates. If we started sending the users language to YouTube, we would need to add support for parsing those relative timestamps in every single language that YouTube supports. Additionally extracting durations on the shorts tab and in the future (I've written the code just not created a pull request for it as it would have caused code conflicts at the time) views and published dates in playlists also all rely on localised strings as that's the only place we can get that information from.

TL;DR always requesting en-US from YouTube makes our life significantly easier, sending anything else would dramatically increase the complexity.

**Comment by absidue** [comment]
Your assumptions in 3 and 4 are completely incorrect.
The wrong audio track being selected is literally just because YouTube returns them all in the response and unless you use a FreeTube version that has support for them, it just selects the first one. The order YouTube returns them in is not affected by us sending en-US, for videos with english as the default language, the english audio formats tend to appear last in the response.

**Comment by absidue** [comment]
duplicate of #1133

**Comment by hollownights** [comment]
> Unfortunately we can't do that yet, as for things like video lists (channels, subscriptions, trending, search results, watch next feed) only gives us relative published dates. If we started sending the users language to YouTube, we would need to add support for parsing those relative timestamps in every single language that YouTube supports. Additionally extracting durations on the shorts tab and in the future (I've written the code just not created a pull request for it as it would have caused code conflicts at the time) views and published dates in playlists also all rely on localised strings as that's the only place we can get that information from.

@absidue Think I got this part, but...

> TL;DR always requesting en-US from YouTube makes our life significantly easier, sending anything else would dramatically increase the complexity.

... you aren't actually doing that. Like I said, at least 3 different languages (`*`, en-US, system language) are being parsed to YouTube, not just en-US. If there is the need to send en-US to get the elements right, fine, but make sure to only send it. There is no need to use * as a language or to let the system's language go through.

> Your assumptions in 3 and 4 are completely incorrect.

Hmmmm Maybe? "hl" is the parameter to setup subtitles, so at some point it may come in to play.

> duplicate of #1133

It is not. That only deals with URL parameters and content recommendation. Here I'm more focused on privacy.

**Comment by absidue** [comment]
> ... you aren't actually doing that. Like I said, at least 3 different languages (*, en-US, system language) are being parsed to YouTube, not just en-US. If there is the need to send en-US to get the elements right, fine, but make sure to only send it. There is no need to use * as a language or to let the system's language go through.

If you look at the context in the request body sent with every Innertube API request, it has `hl` set to `en` and `gl` set to `US`, YouTube cares about that more than the languages in the header.

If you really care about the headers so much we can take a look at it, but it doesn't influence YouTube's responses as we send the correct stuff in the request body.

(And yes I can already hear you screaming about the privacy of that context data, no it isn't a privacy issue)

**Comment by hollownights** [comment]
> If you look at the context in the request body sent with every Innertube API request, it has `hl` set to `en` and `gl` set to `US`, YouTube cares about that more than the languages in the header.

I didn't say the contrary. 

> If you really care about the headers so much we can take a look at it

I do. Thank you very much for your concern and your availability for looking into such issues of privacy.

> (And yes I can already hear you screaming about the privacy of that context data, no it isn't a privacy issue)

Run [Cover Your Tracks](https://coveryourtracks.eff.org/) in Firefox, Chromium, Chrome, Edge etc (all with default settings and maybe once again with uBlockOrigin installed and with its default settings) and then run it in FreeTube. Then, take a look at the "Detailed Results" for each of them (the "Your Results" section is meaningless today).
